src/sinol_make/commands/run/__init__.py

```diff
@@ -368,48 +377,98 @@ def print_view(output_file=None):
+
+				score_expected = self.config["sinol_expected_scores"][prog]["points"]
+				if score_checksum != score_expected:
+					print(util.bold(util.color_red('Program %s will get %d points (expected %d).' % (prog, score_checksum, score_expected))))
+					exit(1)
 
-			validator_program = self.config["subtasks"][subtask]["validator"].split()[0]
-			programs.append(validator_program)
-		programs = list(set(programs))
+				program = self.config["sinol_expected_scores"][prog]["program"].split()[0]
+				programs.append(program)
+			programs = list(set(programs))
 		compilation_results = self.compile_programs(programs)
 		os.makedirs(self.EXECUTIONS_DIR, exist_ok=True)
 		compiled_commands = []
-		for subtask in self.config["subtasks"]:
-			validator_program = os.path.join(self.EXECUTABLES_DIR, self.config["subtasks"][subtask]["validator"])
-			compiled_commands.append((subtask, validator_program, True))
-		names = list(self.config["subtasks"])
-		results = self.perform_executions(compiled_commands, names, programs, self.args.subtask_report)
-		for subtask in self.config["subtasks"]:
-			passed_groups = []
-			for group in self.config["scores"]:
-				if results[subtask][group] == "OK":
-					passed_groups.append(group)
-			passed_groups.sort()
-			should_pass = self.config["subtasks"][subtask]["groups"]
-			should_pass.sort()
-			if passed_groups != should_pass:
-				print(util.bold(util.color_red('Subtask %s will pass groups %s (expected %s).' % (subtask, passed_groups, should_pass))))
-				exit(1)
+		if suggestions:
+			for program in programs:
+				path = os.path.join(self.EXECUTABLES_DIR, program)
+				compiled_commands.append((program, path, True))
+			names = programs
+			results = self.perform_executions(compiled_commands, names, programs, self.args.expected_scores_report)
+
+			print(util.bold("Suggested expected scores description:"))
+			print("sinol_expected_scores:")
+
+			i = 1
+			new_config = self.config
+			new_config["sinol_expected_scores"] = {}
+
+			for program in programs:
+				print("  %s:" % program)
+				print("    program: %s" % program)
+				print("    expected: {" + ', '.join(results[program].items()) + "}")
+				new_config["sinol_expected_scores"][program] = {
+					"program": program,
+					"expected": results[program],
+					"points": 0
+				}
+
+				points = 0
+				for group, result in results[program].items():
+					if result == "OK":
+						points += self.scores[group]
+				print("    points: %d" % points)
+				new_config["sinol_expected_scores"][program]["points"] = points
+
+			if self.args.apply_suggestions:
+				print("Do you want to apply these suggestions? (y/n) ", end="")
+				choice = input().lower()
+				if choice == "y" or choice == "yes" or choice == "":
+					with open(os.path.join(os.getcwd(), "config.yml"), "w") as f:
+						yaml.dump(new_config, f, default_flow_style=False)
+					print(util.color_green("Suggestions applied."))
+				else:
+					print("Suggestions not applied.")
```

Comment:
While the "ask for confirmation" is a good idea (and the implementation seems to be nice), we first need to think about how the users will use the tool and what is one's expectations with each flags.

I think that the users that use `--apply-suggestion` would use it when they are pretty much sure that the suggestion is at it is, or they want to ignore the whole suggestion mechanism (e.g. on the initial stages of creating a task package).

On the contrary, when used without this flag should serve as a confirmation that e.g. the saved scores is wrong and in that case probably the user would want to see the difference (what was saved vs what should be there).

Having those use cases in mind, it would be reasonable to print "what was saved vs what should be there" if something is wrong (bonus points for first printing all of than and only then exiting) and the `--apply-suggestions` should automatically apply them and print what was changed & how it was changed.

Let me know what you think.

Reply:
Yes,I think that this is a better approach

Reply:
> (bonus points for first printing all of than and only then exiting)

Minor suggestion: that probably can be achieved by first building the YAML that should be, and then comparing it with the existing YAML. Should be easy to handle any edge-cases with this method.

Reply:
It should now work as you described it.